modules/javafx.controls/src/main/java/javafx/scene/chart/CategoryAxis.java

```diff
@@ -362,14 +362,21 @@ private double calculateNewFirstPos(double length, double catSpacing) {
         if (length >= 0) {
             double requiredLengthToDisplay = calculateRequiredSize(side.isVertical(), tickLabelRotation);
             if (requiredLengthToDisplay > length) {
-                // try to rotate the text to increase the density
-                if (side.isHorizontal() && tickLabelRotation != 90) {
-                    tickLabelRotation = 90;
+                if (tickLabelRotation != 90) {
+                    var rotatedRequiredLengthToDisplay = calculateRequiredSize(side.isVertical(), 90);
```

Comment:
I prefer an explicit type here (double, I think) so I don't have to go look at the method to know what type it is without going and finding the method.